honeybadgermpc/ipc.py

```diff
@@ -12,24 +15,48 @@ def __init__(self, queues, config):
 
     async def connect(self):
         # Setup all connections first before sending messages
+
+        # XXX BEGIN temporary hack
+        # ``socket.getaddrinfo``, called in open_connection may fail thus causing an
+        # unhandled exception.
+        # A connection management mechanism is needed to that failed connections are
+        # re-tried etc.
+        # Basically nodes should be capable to join when they try to connect but if late
+        # it should not crash the whole thing, or more precisely prevent other nodes
+        # from participating in the protocol.
+        while True:
+            try:
+                addrinfo_list = [
+                    socket.getaddrinfo(self.config[i].ip, self.config[i].port)[0][4]
+                    for i in range(len(self.queues))
+                ]
+            except socket.gaierror:
+                continue
+            else:
+                break
+        # XXX END temporary hack
+
         streams = [asyncio.open_connection(
-                self.config[i].ip,
-                self.config[i].port
+                addrinfo_list[i][0],
+                addrinfo_list[i][1],
```

Comment:
Not block on a gather here until all connections are established. Instead the individual process loops should attempt the connection